paasta_tools/mesos_tools.py

```diff
@@ -749,7 +749,9 @@ def get_mesos_slaves_grouped_by_attribute(slaves, attribute):
     :returns: a dictionary of the form {'<attribute_value>': [<list of hosts with attribute=attribute_value>]}
               (response can contain multiple 'attribute_value)
     """
-    sorted_slaves = sorted(slaves, key=lambda slave: slave["attributes"].get(attribute))
+    sorted_slaves = sorted(
+        slaves, key=lambda slave: slave["attributes"].get(attribute, "")
+    )
     return {
         key: list(group)
         for key, group in itertools.groupby(
```

Comment:
What about line 756? Looks... the same?

Reply:
groupby should be fine since it does not compare keys with </>.

Reply:
I tested that key could be None.